### DISCUSSION TITLE
How to activate both e-mail and SMS at the same time?

### DISCUSSION THREAD

**Discussion by kemalfaruk** [codeigniter4/shield#932] [Q&A]
How to activate both e-mail and SMS at the same time? So, I want to first activate e-mail and then perform SMS verification with the Twilio library. There is this part in Auth.php: public array  
```php
$actions = [
         'register' => \CodeIgniter\Shield\Authentication\Actions\EmailActivator::class,
         'login' => null,
     ];
```
Should I take action in this part?

**Comment by datamweb** [comment]
Defining New Actions, see https://codeigniter4.github.io/shield/references/authentication/auth_actions/?h=ac#defining-new-actions

**Reply by kemalfaruk** [reply]
Can I use both actions at the same time? So can I do both SMS verification and email verification at the same time? There is e-mail activation here. I already did this and it works. Can I verify both SMS and e-mail respectively?

**Reply by sammyskills** [reply]
Yes, you can.

You can create your own custom action, and then perform both SMS and email verifications inside the action.
